tests/serialization/test_dag_serialization.py

```diff
@@ -448,6 +447,12 @@ def test_deserialization_start_date(self, dag_start_date, task_start_date, expec
         simple_task = dag.task_dict["simple_task"]
         self.assertEqual(simple_task.start_date, expected_task_start_date)
 
+    def test_deserialization_with_dag_context(self):
+        with DAG(dag_id='simple_dag', start_date=datetime(2019, 8, 1, tzinfo=timezone.utc)) as dag:
```

Comment:
Was my suggestion here not valid then @houqp, given you said:


> Depends on how you use the context manager. with DAG(): works as expected, but we have huge DAGs (> 9K loc) in production that would be a lot easier to maintain without that extra indentation introduced with the with block. So we are using DagContext.push_context_managed_dag(dag) directly in the DAG definition. This is what triggered the crash.

I.e. does this actually test the code you need it to?

Reply:
your suggestion is still valid, calling serialization within the with block means the dummy task will be created with a DAG pushed to the top of dag context. it covers the same edge case, it's just not how some of our DAGs are written :)